csxtools/utils.py

```diff
@@ -178,7 +178,10 @@ def get_images_to_3D(images, dtype=None):
 
 def _get_images(header, tag, roi=None):
     t = ttime.time()
-    images = header.db.get_images(header, tag)
+    if isinstance(header, (list, tuple)):
+        images = header[0].data(tag)
```

Comment:
Are we only interested in the first image here?

Reply:
@mrakitin is correct -- we don't want to throw away headers in the case of a list input.

We want to be able to use two (or more generally an arbitrary list of) gain 8 dark images to correct the data and this functionality was broken in the last update. 

from databroker import DataBroker as db
from csxtools.utils import get_fastccd_images

old_scan_no = 85528
slicerator = get_fastccd_images(db[old_scan_no], (db[old_scan_no-1, old_scan_no+1], None, None))

i.e. 85527 & 85529 are dark scan_ids with gain 8.

Reply:
Maybe we can do something like this:
```python
images = []
if isinstance(header, (list, tuple)):
    for i in range(len(header)):
        images.append(header.data(tag))
```
I'm not sure if there needs to be a corresponding update downstream.

Reply:
Corrected the code above.

Reply:
(h.data(tag) for h in header) is not working. It fails at get_images_to_3D function. We need a pims object.